#!/bin/python
PACKAGE = "cr_dai_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("confidence",    int_t,    0, "Disparities with confidence value under this threshold are accepted. Higher confidence threshold means disparities with less confidence are accepted too.", 240,  0, 255)
gen.add("lr_check_threshold", int_t, 0, "Left-right check threshold for left-right, right-left disparity map combine, 0..128",    10, 0,   32)
gen.add("bilateral_sigma",    int_t,    0, "Sigma value for bilateral filter. 0 means disabled. A larger value of the parameter means that farther colors within the pixel neighborhood will be mixed together.",  5, 0, 255)
gen.add("subpixel",    bool_t,    0, "Computes disparity with sub-pixel interpolation (5 fractional bits), suitable for long range.",  False)
gen.add("left_right_check",    bool_t,    0, "Computes and combines disparities in both L-R and R-L directions, and combine them.",  True)
gen.add("extended_disparity",    bool_t,    0, "Disparity range increased from 95 to 190, combined from full resolution and downscaled images. Suitable for short range objects",  False)
gen.add("threshold_max", int_t, 0,"Maximum range to return in mm", 10000, 0, 10000)
gen.add("threshold_min", int_t, 0,"Minimum range to return in mm",  0, 0, 10000)

median_enum = gen.enum([ gen.const("MEDIAN_OFF",      int_t, 0, ""),
           gen.const("KERNEL_3x3",     int_t, 3, ""),
           gen.const("KERNEL_5x5",      int_t, 5, ""),
           gen.const("KERNEL_7x7",      int_t, 7, "")],
           "Median filter mode")
gen.add("median_filter_mode", int_t, 0,"Median mode",  0, 0, 7, edit_method=median_enum)

#gen.add("decimation_factor", int_t, 0,"Decimation factor",  1, 1, 4)
#decimation_enum = gen.enum([ gen.const("PIXEL_SKIPPING",      int_t, 0, ""),
#            gen.const("NON_ZERO_MEDIAN",     int_t, 1, ""),
#            gen.const("NON_ZERO_MEAN",      int_t, 2, "")],
#            "Decimation algorithm type.")
#gen.add("decimation_mode", int_t, 0,"Decimation mode",  0, 0, 3, edit_method=decimation_enum)

exit(gen.generate(PACKAGE, "cr_dai_ros_msgs", "StereoDepth"))